zetaclient/chains/base/confirmation.go

```diff
@@ -50,28 +60,57 @@ func (ob *Observer) IsBlockConfirmedForOutboundFast(blockNumber uint64) bool {
 	return isBlockConfirmed(blockNumber, confirmation, lastBlock)
 }
 
+// IsInboundEligibleForFastConfirmation determines if given inbound vote message is eligible for fast confirmation.
+func (ob *Observer) IsInboundEligibleForFastConfirmation(
+	ctx context.Context,
+	msg *crosschaintypes.MsgVoteInbound,
+) (bool, error) {
+	// check if fast confirmation is enabled
+	if !ob.ChainParams().IsInboundFastConfirmationEnabled() {
+		return false, nil
+	}
+
+	// check eligibility
+	if !msg.EligibleForFastConfirmation() {
+		return false, nil
+	}
+
+	// query liquidity cap for asset
+	chainID := msg.SenderChainId
+	fCoins, err := ob.zetacoreClient.GetForeignCoinsFromAsset(ctx, chainID, ethcommon.HexToAddress(msg.Asset))
+	if err != nil {
+		return false, errors.Wrapf(err, "unable to get foreign coins for asset %s on chain %d", msg.Asset, chainID)
+	}
+
+	// ensure the deposit amount does not exceed amount cap
+	fastAmountCap := chains.CalcInboundFastConfirmationAmountCap(chainID, fCoins.LiquidityCap)
+	if msg.Amount.GT(fastAmountCap) {
+		return false, nil
+	}
+
+	return true, nil
+}
+
 // calcUnscannedBlockRange calculates the unscanned block range [from, end (exclusive)) within given block limit.
 //
 // example 1: given lastBlock =  99, lastScanned = 90, confirmation = 10, then no unscanned block
 // example 2: given lastBlock = 100, lastScanned = 90, confirmation = 10, then 1 unscanned block (block 91)
 func calcUnscannedBlockRange(lastBlock, lastScanned, confirmation, blockLimit uint64) (from uint64, end uint64) {
 	// got unscanned blocks or not?
+	// returning same values to indicate no unscanned block
+	nextBlock := lastScanned + 1
 	if lastBlock < lastScanned+confirmation {
-		return 0, 0
+		return nextBlock, nextBlock
```

Comment:
Why this change here?

Reply:
The pair of `(0, 0)` gives no info about `where we are now?` Returning `(nextBlock, nextBlock)` tells the caller the next block to scan.